docs/dev_guides/settings_validation.md

```diff
@@ -0,0 +1,142 @@
+## Settings Schema Validation
```

Comment:
Should we have a sentence or two to explain at a high level what the validators are and why they exist?

For example, something like:
"A common issue within Splink is users providing invalid settings dictionaries. To prevent this, the settings validator scans through a settings dictionary and provides user-friendly feedback on what needs to be fixed. This includes:
1) The structure of the settings dictionary, see Settings Schema Validation
2) The contents of the settings dictionary, see Settings Validator"

Reply:
Yeah, I think that's sensible!